src/main/java/newbank/server/NewBankClientHandler.java

```
@@ -5,119 +5,55 @@
-      out.println("FAIL: " + String.format("Customer name [%s] already exists.", username));
-    } catch (PasswordInvalidException e) {
-      out.println("FAIL: Specified password does not meet the security requirements.");
-    }
+    initialiseSupportedComamnds();
   }
 
-  private void handleShowAccounts() throws RequestNotAllowedException {
-    checkLoggedIn();
-    out.println(bank.showAccountsFor(customer));
+  // add supported commands here
+  private void initialiseSupportedComamnds() {
+    commands.put("LOGIN", (bank, tokens, customer) -> new LoginCommand(bank, tokens, customer));
+    commands.put(
+        "NEWACCOUNT", (bank, tokens, customer) -> new NewAccountCommand(bank, tokens, customer));
+    commands.put(
+        "REGISTER", (bank, tokens, customer) -> new RegisterCommand(bank, tokens, customer));
+    commands.put(
+        "SHOWMYACCOUNTS",
+        (bank, tokens, customer) -> new ShowAccountsCommand(bank, tokens, customer));
   }
 
-  private void checkLoggedIn() throws RequestNotAllowedException {
-    if (customer == null) {
-      throw new RequestNotAllowedException();
-    }
+  private Command getCommand(final String[] tokens) {
+    return commands.get(tokens[0].toUpperCase()).makeCommand(bank, tokens, customer);
   }
 
   private boolean processRequest(final String request) {
     final String[] tokens = request.split("\\s+");
 
     assert (tokens.length > 0);
 
-    try {
-      switch (tokens[0].toUpperCase()) {
-        case "LOGIN":
-          handleLogin(tokens);
-          break;
-        case "REGISTER":
-          handleRegisterCustomer(tokens);
-          break;
-        case "SHOWMYACCOUNTS":
-          handleShowAccounts();
-          break;
-        case "NEWACCOUNT":
-          handleNewAccount(tokens);
-          break;
-        case "QUIT":
-          out.println("SUCCESS: Good bye.");
-          return false;
-        default:
-          out.println("FAIL: Unknown command.");
```

We've lost this functionality. Now unrecognised commands fail silently.
Would be neat to set a function that does the above as the `Map`'s default, and call `getOrDefault` instead of `get`

---

Well spotted, I had planned to update my PR this morning with this.